src/aioftp/client.py

```diff
@@ -636,6 +645,57 @@ async def connect(self, host, port=DEFAULT_PORT):
         code, info = await self.command(None, "220", "120")
         return info
 
+    async def _send_tls_protection_commands(self) -> None:
+        """
+        :py:func:`asyncio.coroutine`
+
+        Sends the PBSZ and PROT commands as required for TLS connection.
+        """
+        await self.command("PBSZ 0", "200")
+        await self.command("PROT P", "200")
+
+    async def upgrade_to_tls(self, sslcontext: ssl.SSLContext = None) -> None:
+        """
+        :py:func:`asyncio.coroutine`
+
+        Attempts to upgrade the connection to TLS (explicit TLS).
+        Downgrading via the CCC or REIN commands is not supported.  Both the command
+        and data channels will be encrypted after using this command.  You may
+        call this command at any point during the connection, but not every FTP server
+        will allow a connection upgrade after logging in.
+
+        :param sslcontext: custom ssl context
+        :type sslcontext: :py:class:`ssl.SSLContext`
+        """
+        if self.stream.writer.transport.get_extra_info("ssl_object"):
+            return
+
+        self.explicit_tls = True
+
+        try:
+            await self.command("AUTH TLS", "234")
+        except errors.StatusCodeError:
+            raise errors.TLSError("Explicit TLS is not supported on this server") from None
```

Comment:
No need to wrap status code error in another. More to say, we are loosing the actual response from server.